src/modules/commander/ManualControl.cpp

```diff
@@ -0,0 +1,194 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2021 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "ManualControl.hpp"
+#include <drivers/drv_hrt.h>
+
+using namespace time_literals;
+
+enum OverrideBits {
```

Comment:
Why not `enum class`?

Reply:
No reason, good suggestion. It was added like that:
https://github.com/PX4/PX4-Autopilot/pull/13618/files#diff-33729b6612e70db2803e6b18b64a47c71e70eaf1d6b69fd233a34b3b0408a806R261
I'm changing it 👍

Reply:
Oh wait there is a reason, It gets used in combination with a bitwise and not direct comparison. Would you define the bitwise and operator in the class or how can it be done easily? Holding back with the change because of lack of good ideas to solve this.

Reply:
It gets used in combination with a "bitwise and" not in a direct comparison. Would you define the bitwise and operator in the class or how can it be done more easily?

That's probably also why it was added in this way:
https://github.com/PX4/PX4-Autopilot/pull/13618/files#diff-33729b6612e70db2803e6b18b64a47c71e70eaf1d6b69fd233a34b3b0408a806R261